### PR TITLE
(maint) fix config migration

### PR DESCRIPTION
**Pull request by wkalt**
Without this we were parsing the subname to get the location of the hsqldb data
directory so it can be migrated. In the default case though, this resides in the vardir, which is
migrated a moment before.

This patch changes things so that we only attempt to migrate that directory if
the original vardir still exists, which is a signal that the vardir is
nonstandard, but the subname still points to the old one (since it's still the
default value.)

### CODE REVIEW

**Comment by wkalt** [comment]
@senior this is sort of horrible, but I think it might do the trick.

**Comment by ajroetker** [comment]
>Without this we were parsing the subname to get the location of the hsqldb data
>directory so it can be migrated. In the default case though, this resides in the vardir, which is
>migrated a moment before.

>This patch changes things so that we only attempt to migrate that directory if
Which is "that" directory?
>the original vardir still exists, which is a signal that the vardir is
"signal that the user had changed the vardir but not the hsqldb subname"
>nonstandard, but the subname still points to the old one (since it's still the
>default value.)
The commit message and PR message use a lot of pronouns that kind of made this hard to parse without a few rereads.

**Comment by wkalt** [comment]
@ajroetker "that directory" is the hsqldb data directory.

File: resources/ext/cli/config-migration [hunk 1]
```diff
@@ -236,13 +238,18 @@ for ini_file in "${ini_files[@]}"; do
...
-                  new_value="${new_value#file:}"
-                  new_value="${new_value%/db;hsqldb.tx=mvcc;sql.syntax_pgs=true}"
-                  echo "Copying over the contents of ${orig_value} to ${new_value}"
-                  migrate "$orig_value" "$new_value"
-                  rm -rf "$orig_value"
+                  orig_hsql_path="${orig_value#file:}"
+                  orig_hsql_dir="${orig_hsql_path%/db;hsqldb.tx=mvcc;sql.syntax_pgs=true}"
+                  new_hsql_path="${new_value#file:}"
+                  new_hsql_dir="${new_hsql_path%/db;hsqldb.tx=mvcc;sql.syntax_pgs=true}"
+                  ## if the default vardir still exists, then the vardir was
```
Comment (ajroetker) [inline resources/ext/cli/config-migration]: s/vardir/vardir config setting

Comment (ajroetker) [inline resources/ext/cli/config-migration]: and therefore was not migrated

File: resources/ext/cli/config-migration [hunk 2]
```diff
@@ -236,13 +238,18 @@ for ini_file in "${ini_files[@]}"; do
...
+                  ## changed, but the subname still points to the default
+                  ## vardir.
```
Comment (ajroetker) [inline resources/ext/cli/config-migration]: default vardir for PuppetDB 2.x

File: resources/ext/cli/config-migration [hunk 3]
```diff
@@ -236,13 +238,18 @@ for ini_file in "${ini_files[@]}"; do
...
+                  if [ -e "/var/lib/puppetdb" ] ; then
+                    echo "Copying over the contents of ${orig_hsql_dir} to ${new_hsql_dir}"
```
Comment (ajroetker) [inline resources/ext/cli/config-migration]: What happens if we find the vardir and subname settings in the opposite order while running the script?

Comment (ajroetker) [inline resources/ext/cli/config-migration]: Nevermind we're just looping so that wouldn't happen

Comment (ajroetker) [inline resources/ext/cli/config-migration]: Why don't we just run the subname migration first in the list?

File: resources/ext/cli/config-migration (hunk 1 above)
Comment (wkalt) [inline resources/ext/cli/config-migration]: it's not the vardir config setting, it's the actual vardir

File: resources/ext/cli/config-migration (hunk 3 above)
Comment (ajroetker) [inline resources/ext/cli/config-migration]: No we can't do that because the vardir migration will `mv` the old vardir on top of the new vardir, deleting the `mv`d hsqldb.

Comment (ajroetker) [inline resources/ext/cli/config-migration]: Can we just check if the old_hsql_dir exists instead of the absolute path?

File: resources/ext/cli/config-migration (hunk 1 above)
Comment (ajroetker) [inline resources/ext/cli/config-migration]: The second `vardir` in this line is not though
